docs/tutorials/clustering/autogmm.ipynb

```diff
@@ -0,0 +1,210 @@
+    "\n",
+    "# Synthetic data\n",
+    "\n",
+    "# Dim 1\n",
+    "class_1 = np.random.randn(150, 1)\n",
+    "class_2 = 2 + np.random.randn(150, 1)\n",
+    "dim_1 = np.vstack((class_1, class_2))\n",
+    "\n",
+    "# Dim 2\n",
+    "class_3 = np.random.randn(150, 1)\n",
+    "class_4 = 2 + np.random.randn(150, 1)\n",
+    "dim_2 = np.vstack((class_3, class_4))\n",
+    "\n",
+    "X = np.hstack((dim_1, dim_2))\n",
+    "\n",
+    "# Labels\n",
+    "label_1 = np.zeros((150, 1))\n",
+    "label_2 = 1 + label_1\n",
+    "\n",
+    "c = np.vstack((label_1, label_2)).reshape(300,)\n",
+    "\n",
+    "# Fit model\n",
+    "autogmm_ = AutoGMMCluster(affinity='all',linkage='all',covariance_type='all')\n",
+    "\n",
+    "# Estimated Labels\n",
+    "c_hat_autogmm = autogmm_.fit_predict(X,c)"
+   ]
+  },
+  {
+   "source": [
+    "We can compare our method to the existing implementation of GMM in Sklearn. Our method expands upon the existing Sklearn framework by allowing the user to automatically find the optimal parameters for a Gaussian mixture model and achieve the best clustering possible. In particular, the ideal number of components `n_components` is output by AutoGMM. If we create a GMM model with our parameters using Sklearn, we will see an optimal fit."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn import mixture\n",
+    "from sklearn.metrics import confusion_matrix\n",
+    "from scipy.optimize import linear_sum_assignment\n",
+    "\n",
+    "# Ideal parameters from AutoGMM\n",
+    "n_comp = autogmm_.n_components_\n",
+    "\n",
+    "cov = autogmm_.covariance_type_\n",
+    "\n",
+    "# Have to provide exact number of optimum components apriori\n",
+    "gmm_ = mixture.GaussianMixture(n_components=n_comp, covariance_type=cov)\n",
+    "gmm_default_ = mixture.GaussianMixture(2)\n",
+    "\n",
+    "# Predicted Labels\n",
+    "c_hat_gmm = gmm_.fit_predict(X)\n",
+    "c_hat_gmm_default = gmm_default_.fit_predict(X)\n",
+    "\n",
+    "# Function to remap labels for direct comparison\n",
+    "def remap_labels(y_true, y_pred):\n",
+    "\n",
+    "    confusion_mat = confusion_matrix(y_true, y_pred)\n",
+    "    row_inds, col_inds = linear_sum_assignment(confusion_mat, maximize=True)\n",
+    "    label_map = dict(zip(col_inds, row_inds))\n",
+    "    remapped_y_pred = np.vectorize(label_map.get)(y_pred)\n",
+    "    \n",
+    "    return remapped_y_pred\n",
+    "\n",
+    "# Remap Predicted labels\n",
+    "c_hat_autogmm = remap_labels(c, c_hat_autogmm)\n",
+    "c_hat_gmm = remap_labels(c, c_hat_gmm)\n",
+    "c_hat_gmm_default = remap_labels(c, c_hat_gmm_default)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Plotting Synthetic clusters\n",
```

Comment:
You probably shouldn't have two consecutive code blocks without any text.  Maybe add a brief bit of markdown introducing the plots.  And, you could make the output of block 3 the ``n_components`` output of Autogmm.